pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -340,3 +352,19 @@ func (srv *ProvisioningSrv) RoutePutAlertRuleGroup(c *models.ReqContext, ag defi
 	}
 	return response.JSON(http.StatusOK, ag)
 }
+
+func determineProvenance(ctx *models.ReqContext) (alerting_models.Provenance, error) {
+	disableProvenanceHeader := ctx.Req.Header.Get(disableProvenanceHeaderName)
```

Comment:
I think I was not super clear with my feedback. The name of the flag already states that it's disabled, so checking true or false is not needed. You can just do something like this:

```golang
	if _, disabled := ctx.Req.Header[disableProvenanceHeaderName]; disabled {
	  return alerting_models.ProvenanceNone
	}
	return alerting_models.ProvenanceAPI
```

Reply:
This makes everything more straight forward.

Reply:
So, what you are saying is that if that header is present, no mater what it contains, we disable provenance, right?

Reply:
👍🏽